app/api/routes/product.py

```diff
@@ -0,0 +1,152 @@
+from fastapi import APIRouter, HTTPException, Depends,Query
+from uuid import UUID, uuid4
+from typing import Dict, Optional
+from pydantic import BaseModel, Field
+from dependencies import get_current_time,get_current_admin,products_db
+from models import Product,ProductCreate,ProductUpdate,ProductSearchParams
+
+
+
+router = APIRouter()
+    
+"""Create a Product"""
+@router.post("/products/", status_code=201)
+async def create_product(product: ProductCreate, admin: bool = Depends(get_current_admin)):
+    
+    if product.name.lower() in products_db:
+        raise HTTPException(status_code=400, detail=f"Product name '{product.name}' already exists. Please use a unique name.")
+    
+    try:
+        new_product = Product(
+            name=product.name,
+            description=product.description,
+            price=product.price,
+            stock=product.stock,
+            is_available=product.is_available
+        )
+        products_db[new_product.id] = new_product.to_dict()
+        return new_product.to_dict()
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
+    
+"""Get the Product by ID"""
+@router.get("/products/{product_id}",status_code=200 , response_model=Product)
+async def get_product_details(product_id: UUID, admin: bool = Depends(get_current_admin)):
+    try:
+        product_uuid = UUID(product_id)
+        product = products_db.get(product_id)
+        if not product:
+            raise HTTPException(status_code=404, detail=f"Product with ID {product_id} not found.")
+    except ValueError:
+        raise HTTPException(status_code=400,detail="Product ID must be a valid integer." )
+    
+    return product
+
+"""Update products by the ID"""
+@router.put("/products/{product_id}", status_code=200 ,response_model=Product)
+async def update_product(product_id: UUID,update_data: ProductUpdate,admin: bool = Depends(get_current_admin)):
+    
+    try:
+        product = products_db.get(product_id)
+        raise HTTPException(status_code=404,detail=f"Product with ID {product_id} not found.")
+        if update_data.name:
+            if any(existing_product['name'].lower() == update_data.name.lower() for existing_product in products_db.values()):
+                raise HTTPException(status_code=400,detail=f"Product name '{update_data.name}' already exists. Please use a unique name.")
+        product.name = update_data.name
+        if update_data.price is not None:
+            if update_data.price < 0:
+                raise HTTPException(status_code=400,detail="Price must be a non-negative number.")
```

Comment:
This exception is already exist on update product model, remove it

Reply:
Done